jetty-core/jetty-security/src/main/java/org/eclipse/jetty/security/AnyUserLoginService.java

```diff
@@ -0,0 +1,111 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security;
+
+import java.util.function.Function;
+import javax.security.auth.Subject;
+
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Session;
+
+/**
+ * A {@link LoginService} which allows unknown users to be authenticated.
```

Comment:
Can you please expand on why this `LoginService` would be useful with an example usage?

Basically this javadoc says: "this is a LoginService that just authenticates users without checking credentials", which seems like a security issue, so one more paragraph explaining that this class should not be used blindly, but perhaps in conjunction with something else?